test/upgrade/prober/verify.go

```diff
@@ -51,7 +52,11 @@ func (p *prober) Verify(ctx context.Context) ([]error, int) {
 	}
 	errs := make([]error, 0)
 	for _, t := range report.Thrown {
-		errs = append(errs, errors.New(t))
+		if p.config.IgnoreDuplicate && strings.HasPrefix(t, "DUPLICATE:") {
```

Comment:
I really don't like checking the type of error by looking at string prefix. That's wobbly code, that easily can break if someone decides to change messages.

See, my general message about idea on how to do it differently.

Reply:
Changed as suggested.